src/zkp/pifac.rs

```diff
@@ -469,13 +462,9 @@ mod tests {
                 &mut rng2,
             )?;
 
-            assert!(mixed_proof
-                .verify(mixed_input, &(), &mut transcript())
-                .is_err());
+            assert!(mixed_proof.verify(input, &(), &mut transcript()).is_err());
```

Comment:
Again, there are two failure modes here: the factors don't match the modulus, and one of them is too small. At the very least the documentation needs to explain this.